pwa/src/components/translations/translationForm.tsx

```diff
@@ -2,96 +2,95 @@ import * as React from "react";
 import {
   GenericInputComponent,
   Card,
-  Alert
+  Spinner
 }
   from "@conductionnl/nl-design-system/lib";
-import {Link} from "gatsby";
-import Spinner from "../common/spinner";
-import FlashMessage from 'react-flash-message';
-import {navigate} from "gatsby-link";
+import { Link } from "gatsby";
 import LoadingOverlay from "../loadingOverlay/loadingOverlay";
+import APIService from "../../apiService/apiService";
+import APIContext from "../../apiService/apiContext";
 
 interface TranslationFormProps {
-  id: string,
+  id?: string,
+  tableName?: string
 }
 
-export const TranslationForm: React.FC<TranslationFormProps> = ({id}) => {
-  const [context, setContext] = React.useState(null);
+export const TranslationForm: React.FC<TranslationFormProps> = ({ id, tableName }) => {
   const [showSpinner, setShowSpinner] = React.useState<boolean>(false);
   const [loadingOverlay, setLoadingOverlay] = React.useState<boolean>(false);
-  const [alert, setAlert] = React.useState<any>(null);
   const [translation, setTranslation] = React.useState<any>(null);
-  const title: string = (id === "new") ? "Create Translation" : "Edit Translation"
+  const title: string = (tableName && tableName !== 'new' && !id) ? "Create translation" : (tableName && id) ? "Edit translation" : (tableName && tableName === 'new') ? "Create table" : null
```

Comment:
By separating the two forms, we wouldn't need complex (and very hard to manage and build upon) checks like these.